src/Descope.vue

```diff
@@ -10,79 +11,61 @@
 			:telemetryKey.attr="telemetryKey"
 			:redirect-url="redirectUrl"
 			:auto-focus="autoFocus"
+			:errorTransformer.prop="errorTransformer"
 			@success="onSuccess"
 			@error="onError"
-			ref="webComponentRef"
 		/>
 	</div>
 </template>
 
-<script lang="ts">
-import DescopeWc from '@descope/web-component';
+<script setup lang="ts">
+import DescopeWcClass from '@descope/web-component';
 import { useOptions, useDescope } from './hooks';
 import { baseHeaders } from './constants';
 import { RequestConfig } from '@descope/core-js-sdk';
-import { SetupContext } from 'vue';
 
-DescopeWc.sdkConfigOverrides = { baseHeaders };
+DescopeWcClass.sdkConfigOverrides = { baseHeaders };
 
-export default {
-	// eslint-disable-next-line vue/multi-word-component-names
-	name: 'Descope',
-	props: {
-		flowId: {
-			type: String,
-			required: true
-		},
-		tenant: {
-			type: String
-		},
-		theme: {
-			type: String
-		},
-		debug: {
-			type: Boolean
-		},
-		telemetryKey: {
-			type: String
-		},
-		redirectUrl: {
-			type: String
-		},
-		autoFocus: {
-			type: Boolean
-		},
-		errorTransformer: {
-			type: Function
-		}
+defineProps({
+	flowId: {
+		type: String,
+		required: true
 	},
-	emits: ['success', 'error'],
-	setup(_: unknown, { emit }: SetupContext) {
-		const { projectId, baseUrl, sessionTokenViaCookie } = useOptions();
-		const sdk = useDescope();
-
-		const onSuccess = async (e: CustomEvent) => {
-			// Note: We need to emit AFTER the afterRequest hook has been called, but for
-			// an unknown reason, the emit is not called if we await the hook.
-			emit('success', e);
-			await sdk.httpClient.hooks?.afterRequest?.(
-				{} as RequestConfig,
-				new Response(JSON.stringify(e.detail))
-			);
-		};
-		const onError = (e: Event) => emit('error', e);
-
-		return {
-			projectId,
-			baseUrl,
-			sessionTokenViaCookie,
-			onSuccess,
-			onError
-		};
+	tenant: {
+		type: String
+	},
+	theme: {
+		type: String
+	},
+	debug: {
+		type: Boolean
+	},
+	telemetryKey: {
+		type: String
 	},
-	mounted() {
-		// Set the errorTransformer function on the web component ref
-		this.$refs.webComponentRef.errorTransformer = this.errorTransformer;
+	redirectUrl: {
+		type: String
+	},
+	autoFocus: {
+		type: Boolean
+	},
+	errorTransformer: {
+		type: Function
 	}
+});
+const emit = defineEmits(['success', 'error']);
+const { projectId, baseUrl, sessionTokenViaCookie } = useOptions();
```

Comment:
There is a lint issue here, let's make sure this not marked as warning, I would expect CI to be fail in such cases.

Reply:
This is for unused attribute, I removed it
Let's merge this PR and change the lint config later on